### PR TITLE
Fix/brizo secretstore

### PR DESCRIPTION
**Pull request by aabdulwahed**
Related Issues: [#291](https://github.com/oceanprotocol/ocean/issues/291)
closes [#444](https://github.com/oceanprotocol/keeper-contracts/issues/444)

### CODE REVIEW

File: contracts/conditions/AccessSecretStoreCondition.sol [hunk 1]
```diff
@@ -92,7 +97,15 @@ contract AccessSecretStoreCondition is Condition, ISecretStore {
         external view
         returns(bool permissionGranted)
     {
-        return documentPermissions[_documentId][_grantee];
+        //return true if the DID provider
+        if(agreementStoreManager.isAgreementDIDProvider
+            (
+                documentPermissions[_documentId].agreementId,
+                msg.sender
```
Comment (ssallam) [inline contracts/conditions/AccessSecretStoreCondition.sol]: This is not correct. What is `msg.sender` when the secret store is making the call?

Comment (ssallam) [reply contracts/conditions/AccessSecretStoreCondition.sol]: Should be checking if `_grantee` is a provider.

File: test/unit/conditions/AccessSecretStoreCondition.Test.js [hunk 2]
```diff
@@ -259,6 +261,49 @@ contract('AccessSecretStoreCondition constructor', (accounts) => {
...
+            )
+
+            await accessSecretStoreCondition.fulfill(nonce, documentId, grantee)
+
+            assert.strictEqual(
+                await accessSecretStoreCondition.checkPermissions(
+                    grantee,
+                    documentId,
+                    {
+                        from: DIDProvider
```
Comment (ssallam) [inline test/unit/conditions/AccessSecretStoreCondition.Test.js]: Please see comment above regarding the use of `msg.sender` in `checkPermissions`. This also needs to change, grantee here should be the DIDProvider

File: test/unit/conditions/AccessSecretStoreCondition.Test.js [hunk 3]
```diff
@@ -259,6 +261,49 @@ contract('AccessSecretStoreCondition constructor', (accounts) => {
...
+            )
+
+            await accessSecretStoreCondition.fulfill(nonce, documentId, grantee)
+
+            assert.strictEqual(
+                await accessSecretStoreCondition.checkPermissions(
+                    DIDProvider,
+                    documentId,
+                    {
+                        from: DIDProvider
```
Comment (ssallam) [inline test/unit/conditions/AccessSecretStoreCondition.Test.js]: 'from' does not matter here, can be removed, but it's not really an issue.

File: contracts/conditions/AccessSecretStoreCondition.sol [hunk 4]
```diff
@@ -92,7 +97,15 @@ contract AccessSecretStoreCondition is Condition, ISecretStore {
         external view
         returns(bool permissionGranted)
     {
-        return documentPermissions[_documentId][_grantee];
+        //return true if the DID provider
+        if(agreementStoreManager.isAgreementDIDProvider
```
Comment (H34D) [inline contracts/conditions/AccessSecretStoreCondition.sol]: can we please have a proper formatting for this block? this is dangerous.

Comment (aabdulwahed) [reply contracts/conditions/AccessSecretStoreCondition.sol]: what you mean by proper formatting ? you mean styling ?

Comment (H34D) [reply contracts/conditions/AccessSecretStoreCondition.sol]: correct

File: test/unit/conditions/AccessSecretStoreCondition.Test.js [hunk 5]
```diff
@@ -259,6 +261,46 @@ contract('AccessSecretStoreCondition constructor', (accounts) => {
         })
     })
     describe('check permissions', () => {
+        it('should grant permission in case of DID provider', async () => {
+            const { DIDProvider } = await setupTest({ registerDID: true })
+            const nonce = constants.bytes32.one
```
Comment (H34D) [inline test/unit/conditions/AccessSecretStoreCondition.Test.js]: We need to remove the term `nonce` here. It is misleading. This is not concern of this PR therefore i opened an issue: https://github.com/oceanprotocol/keeper-contracts/issues/446

Comment (aabdulwahed) [reply test/unit/conditions/AccessSecretStoreCondition.Test.js]: okay, thanks

File: contracts/conditions/AccessSecretStoreCondition.sol (hunk 4 above)
Comment (aabdulwahed) [reply contracts/conditions/AccessSecretStoreCondition.sol]: please check it again.

Comment (H34D) [reply contracts/conditions/AccessSecretStoreCondition.sol]: better, thanks

**Pull request merged**